LinqToRest.Test/QueryProviderTest.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using Messerli.LinqToRest.Test.Stub;
+using Messerli.QueryProvider;
+using Messerli.ServerCommunication;
+using NSubstitute;
+using Xunit;
+
+namespace Messerli.LinqToRest.Test
+{
+    public class QueryProviderTest
+    {
+        [Fact]
+        public void ReturnsRestQuery()
+        {
+            var serviceUri = ResolveServiceUri();
+
+            var resourceRetriever = ResolveResourceRetriever();
+            var objectResolver = ResolveObjectResolver();
+            var queryBinderFactory = ResolveQueryBinderFactory();
+
+
+            var queryProvider = new QueryProvider(resourceRetriever, queryBinderFactory, serviceUri);
+            var query = new Query<ClassWithQueryableMember>(queryProvider);

```

Comment:
Could we put all code up to the actual query creation in a function with a signature like `Query CreateQuery<T>()` which we could use in other tests as well?